pkg/visitors/assemble.go

```diff
@@ -253,6 +267,9 @@ func (v *Assemble) Visit(f uefi.Firmware) error {
 
 		// Fix up the header
 		err = f.GenSecHeader()
+		if f.Header.ExtendedSize > 0xFFFFFF {
```

Comment:
Is this where the spec is not clear? It doesn't say anything about sections >=16MiB, which is the case where a section can contain an encapsulated section which is compressed.

Reply:
The spec is wrong because it says this:
The EFI_COMMON_SECTION_HEADER2 structure is only used if the section is too large to
be described using EFI_COMMON_SECTION_HEADER . Large sections using
EFI_COMMON_SECTION_HEADER2 can only exist in a file using
EFI_FFS_FILE_HEADER2 , the FFS_ATTRIB_LARGE_FILE attribute in the file header is
set.